ruma-api-macros/src/api/metadata.rs

```diff
@@ -61,6 +61,12 @@ impl TryFrom<RawMetadata> for Metadata {
                 },
                 "path" => match expr {
                     Expr::Lit(ExprLit { lit: Lit::Str(literal), .. }) => {
+                        if !literal.value().is_ascii() {
+                            return Err(syn::Error::new_spanned(
+                                literal,
+                                "path may only contain printable ASCII characters",
```

Comment:
`is_ascii` checks whether characters are ASCII, not whether they're *printable* ASCII. The latter is probably the check we want though.